proof/refine/AARCH64/Bits_R.thy

```diff
@@ -56,8 +56,11 @@ lemma isCap_simps:
   "isASIDControlCap w = (w = ASIDControlCap)"
   "isASIDPoolCap w = (\<exists>v0 v1. w = ASIDPoolCap v0 v1)"
   "isVCPUCap w = (\<exists>v. w = VCPUCap v)"
+  "isSGISignalCap w = (\<exists>irq targets. w = SGISignalCap irq targets)"
   by (auto simp: isCap_defs split: capability.splits arch_capability.splits)
 
+lemmas isCap_simps = isCap_simps' isArchSGISignalCap_def
```

Comment:
confused why this manoeuvre was needed

Reply:
`isArchSGISignalCap` is not a "normal" isCap predicate from the data type definitions (it's defined manually), but it is used like one, that's why it's added to the standard simp set for these.

Reply:
I thought I had posted a follow-up to this. If there is a nonsensical answer on another thread, then that's probably where it went.

The block above spells out the unfolded existential for the other `isCap_simps`. For this one that's just the `_def`, so I didn't want to repeat the statement. In terms of arch split, the `isArchSGISignalCap_def` should go into the arch part, of course.

Reply:
That might benefit from a comment. Corey was already confused by the difference between isCap_simps and isCap_defs.

Reply:
ah, yes, I didn't even think of that distinction. Happy to add.